blocks/collapse/collapse.js

```diff
@@ -1,122 +1,32 @@
-                <div class="button-link button-link--link">
-                  <button class="button-link--icon" title="Mehr Informationen ${item.title.uncapitalizedLabel}" aria-label="Mehr Informationen ${item.title.uncapitalizedLabel}">
-                    <span class="button-link__content">
-                      <span>Mehr Informationen ${item.title.uncapitalizedLabel}</span>
-                    </span>
-    
-                    <span class="icon icon-expand-more"></span>
-                  </button>
-                </div>
-              </div>
-            </div>
-
-          </div>
-        </div>
-      </div>
-    </div>
-  `).join('');
-}
-
-function addCollapseListener(collapse) {
-  const expandable = collapse.querySelector('.text-block__expandable-area');
-  expandable.style.transition = 'height 500ms ease-in-out';
-
-  const textBlock = expandable.querySelector('.text');
-  const textBlockButton = expandable.querySelector('.text-block__button');
-
-  collapse.addEventListener('click', (event) => {
-    const toggle = event.target.closest('.button-toggle');
-    if (toggle) {
-      const isExpanded = toggle.classList.contains('button-toggle--expanded');
-      const isCollapsed = toggle.classList.contains('button-toggle--collapsed');
-
-      collapse.classList.toggle('text-block--expanded', !isExpanded);
-      toggle.classList.toggle('button-toggle--expanded', !isExpanded);
-      toggle.classList.toggle('button-toggle--collapsed', !isCollapsed);
-
-      const button = toggle.querySelector('button');
-      const buttonContent = button.querySelector('.button-link__content');
-      const newLabel = isExpanded ? toggle.dataset.collapsedLabel : toggle.dataset.expandedLabel;
-      button.setAttribute('aria-label', newLabel);
-      button.setAttribute('title', newLabel);
-      buttonContent.textContent = newLabel;
+function addCollapseListener(collapseBlock) {
+  const expandableText = collapseBlock.querySelector('div:nth-child(2)');
+  const collapsibleSection = collapseBlock.parentElement.parentElement;
+  const expandButton = collapseBlock.querySelector('div:nth-child(3)');
+  const collapseButton = collapseBlock.querySelector('div:nth-child(4)');
+  const arrowUpElement = document.createElement('div');
+  if (!expandButton && !collapseButton) return;
+  arrowUpElement.classList.add('arrow', 'down');
+  expandButton.querySelector('div').appendChild(arrowUpElement);
+  const arrowDownElement = document.createElement('div');
+  arrowDownElement.classList.add('arrow', 'up');
+  collapseButton.querySelector('div').appendChild(arrowDownElement);
+
+  expandButton.addEventListener('click', () => {
+    const height = collapseBlock.querySelector('div:nth-child(2) > div').clientHeight
+        + collapseButton.getBoundingClientRect().height
+        + parseFloat(getComputedStyle(expandableText).paddingBottom);
+    expandableText.style.height = `${height}px`;
```

Comment:
Could we simplify this by just toggling the height ? For e.g. like here https://jsfiddle.net/nd0s2vbw/

Reply:
In here the height needs to change because the animation that expands the text is based on height changes https://github.com/hlxsites/zeiss/pull/73/files#diff-8363bad8a8444eb5bf8b7a2a52e26ae09d2c83b0e13dc442fe01830dbfd2c563R17

Reply:
I tried to apply at least the height: 0 to the .collapsed-text .collapse > div:nth-child(2) class, but not sure why that doesn't work, I will try to check what I am missing